client/heartbeatstop.go

```diff
@@ -31,13 +32,13 @@ func newHeartbeatStop(
 	shutdownCh chan struct{}) *heartbeatStop {
 
 	h := &heartbeatStop{
-		startupGrace:  time.Now().Add(timeout),
-		allocInterval: make(map[string]time.Duration),
-		allocHookCh:   make(chan *structs.Allocation),
-		getRunner:     getRunner,
-		logger:        logger,
-		shutdownCh:    shutdownCh,
-		lock:          &sync.RWMutex{},
+		startupGrace: time.Now().Add(timeout),
+		allocHookCh:  make(chan *structs.Allocation, 10),
```

Comment:
During client restart it's possible to get a bunch of these all at once, so having a larger buffer gives us some room to avoid putting backpressure on the allocrunner setup.